### PR TITLE
qownnotes: fix build for macOS

### PR DESCRIPTION
**Pull request by pbek**
This should fix the build on macOS, where there is no `xvfb-run`. Or is this only an issue on `aarch64 darwin`, @wineee?
(See https://github.com/pbek/QOwnNotes/issues/2912#issuecomment-2525284466)

But https://github.com/NixOS/nixpkgs/blob/22c3f2cf41a0e70184334a958e6b124fb0ce3e01/pkgs/tools/misc/xvfb-run/default.nix#L60 told me there is no `xvfb-run`.

@sfatula, maybe you want to test this build?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/office/qownnotes/default.nix [hunk 1]
```diff
@@ -52,13 +51,15 @@ stdenv.mkDerivation {
...
-  postInstall = ''
-    installShellCompletion --cmd ${appname} \
-      --bash <(xvfb-run $out/bin/${appname} --completion bash) \
-      --fish <(xvfb-run $out/bin/${appname} --completion fish)
-    installShellCompletion --cmd ${pname} \
-      --bash <(xvfb-run $out/bin/${appname} --completion bash) \
-      --fish <(xvfb-run $out/bin/${appname} --completion fish)
+  postInstall = ""
+    # Install shell completion on Linux (there is no xvfb-run on macOS)
+    + lib.optionalString stdenv.isLinux ''
```
Comment (getchoo) [inline pkgs/applications/office/qownnotes/default.nix:56]:
```suggestion
  # Install shell completion on Linux (there is no xvfb-run on macOS)
  postInstall = lib.optionalString stdenv.isLinux ''
```

**Review by getchoo — COMMENTED** [review]
I think it would also be worth checking if the Darwin executable will spit out the completions without this command -- or if there is an alternative for xquartz as mentioned in the issue

**Comment by getchoo** [comment]
<details>
  <summary>Log</summary>
  
```
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_tl.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_fil.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ceb.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_hi.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_hil.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ur.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_fi.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_el.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_gl.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_no.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_da.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ro.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_sl.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_sk.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_eu.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_sr.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_fa.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ha.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_bg.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_is.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_pa.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_sq.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_he.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_yi.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_th.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_sn.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_km.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_si.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_zu.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_xh.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ms.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_mi.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ga.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_uz.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_vi.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_lv.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ku.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_lt.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_bs.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_mk.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_ko.qm
-- Installing: /nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/share/qt6/translations/QOwnNotes_et.qm
mv: cannot stat '/nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/bin/QOwnNotes.app': No such file or directory
```

</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363327`

---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>qownnotes</li>
  </ul>
</details>

**Comment by pbek** [comment]
@getchoo, so do you have Apple Silicon hardware? That's amazing! Thank you for testing!

> mv: cannot stat '/nix/store/1fq8jcv4ziaf16d5c5a7748jlvgpr2j7-qownnotes-24.11.2/bin/QOwnNotes.app': No such file or directory

Yes, we just saw that in https://github.com/pbek/QOwnNotes/issues/2912#issuecomment-2526373528.

Can the paths in https://github.com/NixOS/nixpkgs/blob/22c3f2cf41a0e70184334a958e6b124fb0ce3e01/pkgs/applications/office/qownnotes/default.nix#L67-L72 be adapted to work for aarch64-darwin?

But why did it ever work on x86_64-darwin? 🤔

**Comment by pbek** [comment]
I played around in a GitHub workflow, the build should work now.
https://github.com/pbek/QOwnNotes/actions/runs/12243377462

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/applications/office/qownnotes/default.nix [hunk 2]
```diff
@@ -52,23 +51,30 @@ stdenv.mkDerivation {
...
+      installShellCompletion --cmd ${pname} \
+        --bash <($out/bin/${appname} --completion bash) \
+        --fish <($out/bin/${appname} --completion fish)
   ''
   # Create a lowercase symlink for Linux
   + lib.optionalString stdenv.hostPlatform.isLinux ''
     ln -s $out/bin/${appname} $out/bin/${pname}
   ''
-  # Wrap application for macOS as lowercase binary
+  # Remame application for macOS as lowercase binary
```
Comment (Aleksanaa) [inline pkgs/applications/office/qownnotes/default.nix:73]:
```suggestion
  # Rename application for macOS as lowercase binary
```

(Also why is this needed?)

Comment (pbek) [reply pkgs/applications/office/qownnotes/default.nix]: Thank you for your typo fix!

> (Also why is this needed?)

I'm not sure. In the original package, there was a step to create a lowercase binary. I was just trying to mimic that again. ️🤷🏻

Also, I don't know why no `.app` file falls out of the build process... Does anyone have experience with Darwin builds on NixOS?

**Comment by pbek** [comment]
Thank you!
